munimap/queue/queue.py

```diff
@@ -141,20 +141,33 @@ def get(self, id):
         with self._get_conn() as conn:
             cursor = conn.execute(self._get, (id,))
             try:
-                priority, obj_buffer, result, status = cursor.next()
+                priority, obj_buffer, result, status = next(cursor)
                 j = Job(
                     id=id,
-                    definition=loads(str(obj_buffer)),
+                    definition=loads(str(obj_buffer.decode())),
                     status=status,
                     priority=priority,
                 )
-                j.result = loads(str(result or 'null'))
+                j.result = loads(str(result.decode() or 'null'))
                 return j
             except StopIteration:
                 return None
 
     def mark_done(self, id, result):
-        obj_buffer = buffer(dumps(result))
+        for x in result:
+          if isinstance(result[x], bytes):
+            result.update({x: result[x].decode()})
+          elif isinstance(x, bytes):
+            result.pop(x)
```

Comment:
Is it a good idea to remove element from the list you are currently iterating over, or will this have unintended side effects?

Reply:
It would be nice to get some feedback on this

Reply:
This is ok. This is just checking if the variable is of instance bytes. If it is, it first needs to be decoded, and then the result is updated, as you can see in the next line. So basically, it is being replaced by a valid decoded value.

Reply:
Thanks for the info. I just noticed that you are iterating over the keys of a dict - not over a list. So this is ok